File: src/Models/Navdata/Fixes/GeoPoint.cs
namespace Net.Vatprc.Uniapi.Models.Navdata.Fixes;

public class GeoPoint : Fix
{
    public GeoPoint(double latitude, double longitude) : base(latitude, longitude)
    {
    }

    public override bool Equals(object? obj)
    {
        return obj is GeoPoint point &&
            Latitude == point.Latitude &&
            Longitude == point.Longitude;
    }

    public override int GetHashCode()
    {
        return base.GetHashCode();
    }
}

File: src/model/navdata/geo_point.rs
use crate::model::navdata::Fix;

#[derive(Debug, Clone, PartialEq)]
pub struct GeoPoint {
    pub latitude: f64,
    pub longitude: f64,
}

impl Fix for GeoPoint {
    fn latitude(&self) -> f64 {
        self.latitude
    }

    fn longitude(&self) -> f64 {
        self.longitude
    }
}
